docs/design.md

```diff
@@ -109,11 +109,19 @@ A fully-populated example of an `UpgradeConfig` status is included below:
 
 ## Config Managers
 
-The `managed-upgrade-operator` provides a configurable mechanism for retrieving and storing an `UpgradeConfig` 
+The `managed-upgrade-operator` provides a configurable mechanism for retrieving and storing an `UpgradeConfig`
 Custom Resource that can be reconciled against by the `UpgradeConfig` controller.
 
 For more information, see the dedicated section on this topic: [UpgradeConfig Managers](configmanager.md)
 
+## Controllers
+
+The `managed upgrade operator` provided upgrade process revolves around multiple Controllers. Alongside the above mentioned `UpgradeConfig` controller, the `NodeKeeper` controller works simultaneously in an upgrade process towards the state of nodes in the cluster.
+
+The `NodeKeeper` controller keeps a track of state of all the nodes during an upgrade, making sure of any activities around the nodes and acts upon any action required towards maintaining the required state of any node.
```

Comment:
```suggestion
The `NodeKeeper` controller keeps a track of the upgrading worker nodes during an upgrade and seeks to ensure their timely and eventual upgrade.
```

Reply:
Just a couple of suggestions here to help clarify that it's worker nodes specifically.

Reply:
Thanks for the suggestion, I have made changes according to the suggestion.